### PR TITLE
Metaphlan/makedb

### LINKED ISSUE #2954: metaphlan3/make_db
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by LilyAnderssonLee**
## PR checklist

Closes #2954  

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/metaphlan/makedb/meta.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: metaphlan_makedb
+description: Builds a METAPHLAN database
```
Comment (sofstam) [inline modules/nf-core/metaphlan/makedb/meta.yml:2]:
```suggestion
description: Build MetaPhlAn3 database for taxonomic profiling.
```

File: modules/nf-core/metaphlan/makedb/meta.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+keywords:
+  - metaphlan
+  - index
+  - database
+  - latest
```
Comment (sofstam) [inline modules/nf-core/metaphlan/makedb/meta.yml:7]:
```suggestion
  - metagenomics
```

File: modules/nf-core/metaphlan/makedb/meta.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    homepage: https://huttenhower.sph.harvard.edu/metaphlan/
+    documentation: https://github.com/biobakery/MetaPhlAn
+    doi: "10.7554/eLife.65088"
+    licence: ["MIT License"]
+
+output:
+  - db:
+      type: directory
+      description: Output directory containing the indexed METAPHLAN3 database
+      pattern: "*"
```
Comment (sofstam) [inline modules/nf-core/metaphlan/makedb/meta.yml:20]:
```suggestion
      pattern: "*/"
```

**Review by sofstam — CHANGES_REQUESTED** [review]
Nice work!

File: modules/nf-core/metaphlan/makedb/meta.yml (hunk 3 above)
Comment (LilyAnderssonLee) [reply modules/nf-core/metaphlan/makedb/meta.yml]: OK

File: modules/nf-core/metaphlan/makedb/meta.yml (hunk 2 above)
Comment (LilyAnderssonLee) [reply modules/nf-core/metaphlan/makedb/meta.yml]: Great!

File: modules/nf-core/metaphlan/makedb/main.nf [hunk 4]
```diff
@@ -0,0 +1,53 @@
+process METAPHLAN_MAKEDB {
+    label 'process_medium'
+
+    conda "bioconda::metaphlan=4.0.6"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/metaphlan:4.0.6--pyhca03a8a_0' :
+        'quay.io/biocontainers/metaphlan:4.0.6--pyhca03a8a_0' }"
+
+    output:
+    path "metaphlan_db_latest" , emit: db
```
Comment (sofstam) [inline modules/nf-core/metaphlan/makedb/main.nf:10]:
```suggestion
    path "metaphlan_db_latest"      , emit: db
```

File: modules/nf-core/metaphlan/makedb/meta.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    description: Identify clades (phyla to species) present in the metagenome obtained from a microbiome sample and their relative abundance
+    homepage: https://huttenhower.sph.harvard.edu/metaphlan/
+    documentation: https://github.com/biobakery/MetaPhlAn
+    doi: "10.7554/eLife.65088"
+    licence: ["MIT License"]
+
+output:
+  - db:
+      type: directory
+      description: Output directory containing the indexed METAPHLAN3 database
```
Comment (sofstam) [inline modules/nf-core/metaphlan/makedb/meta.yml:19]:
```suggestion
      description: Output directory containing the indexed MetaPhlAn3 database
```

**Review by sofstam — APPROVED** [review]
Two minor suggestions and ready to merge.

**Pull request merged**